src/App.tsx

```diff
@@ -1,61 +1,160 @@
+  const [userId, setUserId] = useState('');
+
+  const [languageError, setlanguageError] = useState(false);
+  const [isTitleError, setTitleError] = useState(false);
+  const [isUserError, setUserError] = useState(false);
+
+  const [visibleTodos, setVisibleTodos] = useState(todoWithUsers);
+
+  let maxId = Math.max(...todoWithUsers.map(todo => todo.id));
+
+  const createTodo = () => {
+    maxId += 1;
+
+    return ({
+      id: maxId,
+      title,
+      completed: false,
+      userId: +userId,
+      user: getUser(+userId),
+    });
+  };
+
+  function checkValue(inputValue: string) {
+    const reg = /[a-zA-Zа-яА-ЯГЄІЇ\d\s]/g;
+    const matches = inputValue.match(reg);
+
+    if (matches === null) {
+      setlanguageError(true);
+    }
+
+    return matches ? matches.join('') : '';
+  }
+
+  type Event = React.ChangeEvent<HTMLInputElement | HTMLSelectElement>;
+
+  const handleField = (e: Event) => {
+    const { value } = e.target;
+
+    if (e.target.id === 'title') {
+      setTitleError(false);
+      setlanguageError(false);
+
+      setTitle(checkValue(value));
+    }
+
+    if (e.target.id === 'select') {
+      setUserId(value);
+      setUserError(false);
+    }
+  };
+
+  const handleNewTodo = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    setTitleError(!title);
+    setUserError(!userId);
+
+    if (title && userId) {
+      const newTodo = createTodo();
+
+      setVisibleTodos([
+        ...visibleTodos,
+        newTodo,
+      ]);
+      setUserId('');
+      setTitle('');
+    }
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        onSubmit={handleNewTodo}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            Title:
+            {' '}
+            <input
+              type="text"
+              id="title"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={title}
+              onChange={e => handleField(e)}
+            />
+            {languageError
+              && (
+                // eslint-disable-next-line max-len
+                <span className="error">The title supports only EN and UA letters and numbers</span>
+              )}
+
+            {isTitleError && (
+              <span className="error">Please enter a title</span>
+            )}
+          </label>
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+          <label>
+            User:
+            {' '}
+            <select
+              data-cy="userSelect"
+              value={userId}
+              id="select"
+              onChange={e => handleField(e)}
```

Comment:
Also will work. Event goes to the arguments by default 
```suggestion
              onChange={handleField}
```

Reply:
thanks for the good review and for your advice